staging/src/k8s.io/api/auditregistration/v1alpha1/types.go

```diff
@@ -196,3 +223,113 @@ type ServiceReference struct {
+
+	// `userGroups` in this selector. A user is considered matching
+	// if it is a member of any of the `userGroups`.
+	// An empty list implies every user group.
+	// +optional
+	UserGroups []string `json:"userGroups,omitempty" protobuf:"bytes,3,rep,name=userGroups"`
+
+	// `verbs` included in this selector.
+	// An empty list implies every verb.
+	// +optional
+	Verbs []string `json:"verbs,omitempty" protobuf:"bytes,4,rep,name=verbs"`
+
+	// Selectors can apply to API resources (such as "pods" or "secrets"),
+	// non-resource URL paths (such as "/api"), or neither, but not both.
+	// If neither is specified, the selector is treated as a default for all URLs.
+
+	// `resources` in this selector. An empty list implies all kinds in all API groups.
+	// +optional
+	Resources []GroupResources `json:"resources,omitempty" protobuf:"bytes,5,rep,name=resources"`
+
+	// `namespaces` in this selector.
+	// The empty string "" matches non-namespaced resources.
+	// An empty list implies every namespace.
+	// Non-namespaced resources will be matched if and only if the empty string is present in the list
+	// +optional
+	Namespaces []string `json:"namespaces,omitempty" protobuf:"bytes,6,rep,name=namespaces"`
+
+	// `nonResourceURLs` is a set of URL paths that should be audited.
+	// *s are allowed, but only as the full, final step in the path, and are delimited by the path separator
+	// Examples:
+	//  "/metrics" - Log requests for apiserver metrics
+	//  "/healthz/*" - Log all health checks
+	// +optional
+	NonResourceURLs []string `json:"nonResourceURLs,omitempty" protobuf:"bytes,7,rep,name=nonResourceURLs"`
+}
+
+// GroupResources represents resource kinds in an API group.
+type GroupResources struct {
+	// `group` is the name of the API group that contains the resources.
+	// The empty string represents the core API group.
+	// +optional
+	Group string `json:"group,omitempty" protobuf:"bytes,1,opt,name=group"`
+
+	// `resources` is a list of resources in this group.
+	//
+	// For example:
+	// 'pods' matches pods.
+	// 'pods/log' matches the log subresource of pods.
+	// '*' matches all resources and their subresources.
+	// 'pods/*' matches all subresources of pods.
+	// 'pods*' would not match any subresources of pods because it is before the path separator.
+	// '*/scale' matches all scale subresources.
+	//
+	// If wildcard is present, the validation rule will ensure resources do not
+	// overlap with each other.
+	//
+	// An empty list implies all resources and subresources in these API groups apply.
+	// +optional
+	Resources []string `json:"resources,omitempty" protobuf:"bytes,2,rep,name=resources"`
+
+	// `objectNames` is a list of resource instance names in this group.
+	// Using this field requires resources to be specified.
+	// An empty list implies that every instance of the resource is matched.
+	// +optional
+	ObjectNames []string `json:"objectNames,omitempty" protobuf:"bytes,3,rep,name=objectNames"`
```

Comment:
Shouldn't this be `ResourceNames` to be consistent with `RBAC`, `AdmissionWebhooks` and the current `AuditPolicy`?

Then copying existing rules into classes can be done easily.